internal/ast/ast.go

```diff
@@ -292,12 +292,17 @@ func (e *NumExpr) String() string {
 	}
 }
 
-// StrExpr is a literal string like "foo".
+// StrExpr is a literal string like "foo" or a regexp constant like /foo/.
 type StrExpr struct {
-	Value string
+	Value  string
+	Regexp bool
 }
 
 func (e *StrExpr) String() string {
+	if e.Regexp {
+		r := &RegExpr{e.Value}
+		return r.String()
```

Comment:
Nit suggestion (untested, but I think it should work):
```suggestion
		return RegExpr.String(e.Value)
```

Reply:
It won't, because the (*RegExpr).String takes a *RegExpr as an argument, not a string.

Reply:
Ah yes, you're right.

Reply:
Building a `RegExpr` object here still feels a little weird -- can we factor out the content of `RegExpr.String` into `formatRegex` and call it from both places?

Reply:
Yeah